compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2146,13 +2146,15 @@ object Types {
   }
 
   case class LazyRef(refFn: () => Type) extends UncachedProxyType with ValueType {
+    private var myRefFn: () => Type = refFn
```

Comment:
I think this would be slightly nicer as `case class LazyRef(private var refFn: () => Type)`, this way you cannot accidentally capture `refFn` outside of the constructor.